impl/httpMetricsManager.go

```diff
@@ -2,24 +2,62 @@ package impl
 
 import (
 	"fmt"
+	"sort"
+	"strings"
 	"time"
 
+	"github.com/satori/go.uuid"
 	"github.com/xiaoyao1991/chukonu/core"
 )
 
 type HttpMetricsManager struct {
 	requestSent int
 	countQueue  chan int
+	records     HttpRecords
+	durationSum DurationSummation
+	numOfError  int
 	// throughputRecord []int
 }
 
+type HttpRecord struct {
+	duration time.Duration
+	id       uuid.UUID
+	status   string
+}
+type HttpRecords []HttpRecord
+
+type DurationSummation struct {
+	sum   time.Duration
+	index int
+}
+
+// ------------sorting interface for struct HttpRecord-----------
+func (slice HttpRecords) Len() int {
+	return len(slice)
+}
+func (slice HttpRecords) Less(i, j int) bool {
+	return slice[i].duration < slice[j].duration
+}
+func (slice HttpRecords) Swap(i, j int) {
+	slice[i], slice[j] = slice[j], slice[i]
+}
+
+//----------------------------------------------------------------
+
 func (m *HttpMetricsManager) RecordRequest(request core.ChukonuRequest) {
 }
 func (m *HttpMetricsManager) RecordResponse(response core.ChukonuResponse) {
+	record := new(HttpRecord)
+	record.duration = response.Duration()
+	record.id = response.ID()
+	record.status = response.Status()
+	m.records = append(m.records, *record)
+	if !strings.Contains(response.Status(), "200") {
+		m.numOfError += 1
+	}
 }
 func (m *HttpMetricsManager) RecordError(err error) {
```

Comment:
`numOfError` should be incremented here instead. I expect this method to be called when an error is returned from `Engine.RunRequest`